Alpaka as 'git subtree' instead of 'git submodule'

tdd11235813 opened the pull request:
Changes Alpaka dependency workflow:
- replaces the git submodule with git subtree 
- new workflow documented in [README.md](https://github.com/tdd11235813/cupla/blob/1836ac1b886a7b797659f8a037dda1ee28203c05/README.md#alpaka-as-git-subtree)

- [x] merge after #104 (is required to work)

Commit:
Removes alpaka git module.

ax3l commented on cuplaConfig.cmake:
```diff
@@ -93,6 +93,11 @@ list(APPEND CMAKE_MODULE_PATH "${_cupla_ROOT_DIR}")
 
 
+# C++11 standard for whole project (this is required, even though Alpaka sets C++11 too)
+set(CMAKE_CXX_STANDARD 11)
```
these variables must not be set in library code but are set by the application using cupla.

Try setting these as target properties from determined `CMAKE_CXX_STANDARD` (with fallbacks to C++11 if unset).

sbastrakov replied:
Agree this variable should not be set like that. Maybe an even easier solution without using `CMAKE_CXX_STANDARD` at all: [example 1](https://github.com/robotology/how-to-export-cpp-library/issues/44), [example 2](https://cliutils.gitlab.io/modern-cmake/chapters/features/cpp11.html). This would effectively set a minimum required version: if a host application is setting higher version in `CMAKE_CXX_STANDARD`, that will be used. If a host application does not set it or sets lower version, C++11 will be used for cupla.

tdd11235813 replied:
yeah, you are absolutely right, will look into that.

ax3l replied:
You can also take some details from openPMD-api in which I try to follow recent CMake best-practices: https://github.com/openPMD/openPMD-api/blob/dev/CMakeLists.txt

sbastrakov commented on README.md:
```diff
@@ -37,7 +37,7 @@ Dependencies
 - **cmake 3.11.0**
-- **alpaka** 
+- **alpaka**
   - is loaded as `git submodule` within **cupla** (see [INSTALL.md](INSTALL.md))
```
Doesn't this also need an update?

Commit:
CUPLA_ADD_EXECUTABLE links cupla target. Target provides c++11.

Commit:
Some stylistic changes.

Commit:
Alpaka as git subtree instead of git submodule.

- no internal git submodule updater, as git subtree is now used
- Updates doc for git subtree feature.
- git subtree details in INSTALL.md

Update to this version requires (git will complain otherwise):
- rm -rf ./alpaka
- rm -rf .git/modules/alpaka

For the record, here is the module unregister procedure (is not needed
for updating):

1. git submodule deinit -f -- alpaka
2. rm -rf .git/modules/alpaka
3. git rm -f alpaka

tdd11235813 commented:
- moved C++11 fix to #104 (so CPU backends might fail now, going to rebase after #104)
- git subtree workflow now documented in `INSTALL.md` (`README.md` is updated as well)
- Removes cupla_ALPAKA_PROVIDER
  - no internal git submodule updater, as git subtree is now used
  - only one find_package(alpaka)
- click on [last commit](https://github.com/ComputationalRadiationPhysics/cupla/pull/103/commits/0b2c0f69864998741ef4c66e68fc965e88848eea) to see that changes

Commit:
Squashed 'alpaka/' content from commit 0a2b6161

git-subtree-dir: alpaka
git-subtree-split: 0a2b616129678606da6f7c23902da28269721f41

Commit:
Merge commit '16b4551075f93cdd71309433b02936a152dfee9f' as 'alpaka'

tdd11235813 commented:
updated alpaka and rebased onto #104 to fix issue.
- cmake alpaka internal provider is back to provide explicit control of alpaka path selection

Note: rebasing with upstream lead git to be confused totally cuz of the submodule vs subtree (it mixed alpaka files with cupla files^^), re-adding subtree from scratch helped

ax3l commented on INSTALL.md:
```diff
@@ -41,3 +38,28 @@ compile an example
+## add subtree (not needed, already cloned with git)
+# git subtree add --prefix alpaka https://github.com/ComputationalRadiationPhysics/alpaka.git develop --squash
+## Update
+git subtree pull --prefix alpaka https://github.com/ComputationalRadiationPhysics/alpaka.git develop --squash
```
Please update with a generic tool user name and e-mail, see https://github.com/ComputationalRadiationPhysics/picongpu/pull/2893

tdd11235813 commented:
yeah @psychocoderHPC also mentioned this offline today. Any suggestions for the mail address? Not sure whether it needs to be a valid one. 
I would like to suggest a more generic one instead of `cupla@hzdr.de`, because it also will affect other [upcoming] (beta-)projects too, and I do not want to create myriads of addresses. Maybe `crp-git@hzdr.de` or something like that? I would add your workflow to the INSTALL.md then.

psychocoderHPC commented:
`cupla@hzdr.de` is fine but let us move to the generic address `crp-git@hzdr.de` and use it in all new projects.

ax3l commented:
Sure, please update your instructions in `INSTALL.md` with it :)

tdd11235813 commented:
ok, see PR #111
